metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/execution/QueueConsumer.java

```diff
@@ -100,63 +98,97 @@ public QueueConsumer(Channel rabbitmqConsumerChannel, String rabbitmqQueueName,
   public void handleDelivery(String consumerTag, Envelope rabbitmqEnvelope,
       AMQP.BasicProperties properties, byte[] body) throws IOException {
     String objectId = new String(body, StandardCharsets.UTF_8);
-    LOGGER.info("WorkflowExecution id: {} received from queue.", objectId);
+    LOGGER.info("workflowExecutionId: {} - Received from queue.", objectId);
+    LOGGER.info("workflowExecutionId: {} - Claiming workflow execution", objectId);
+    Pair<WorkflowExecution, Boolean> workflowExecutionClaimedPair = workflowExecutionMonitor
+        .claimExecution(objectId);
+    WorkflowExecution workflowExecution = workflowExecutionClaimedPair.getLeft();
+    final Boolean workflowClaimed = workflowExecutionClaimedPair.getRight();
 
+    boolean sendAck = true;
+    if (workflowExecution == null) {
+      // This execution no longer exists and we need to ignore it.
+      LOGGER.warn("workflowExecutionId: {} - Was in queue but no longer exists.", objectId);
+    } else if (workflowClaimed.equals(Boolean.TRUE)) {
+      handleClaimedExecution(workflowExecution);
+    } else if (workflowClaimed.equals(Boolean.FALSE) && WorkflowExecutionMonitor.CLAIMABLE_STATUSES
+        .contains(workflowExecution.getWorkflowStatus())) {
+      LOGGER.info(
+          "workflowExecutionId: {} - Send workflow execution identifier back to the queue, it "
+              + "could not be claimed.", workflowExecution.getId());
+      sendAck = false;
+    } else {
+      LOGGER.info("workflowExecutionId: {} - Does not have a claimable status, discarding message",
+          workflowExecution.getId());
+    }
+    if (sendAck) {
+      sendAck(rabbitmqEnvelope, objectId);
+    } else {
+      sendNack(rabbitmqEnvelope, objectId);
```

Comment:
Should this be in a finally block so that we know it is always sent?

Reply:
Wouldn't matter much, but wouldn't hurt having it in a finally block either